registrations/views.py

```diff
@@ -213,3 +213,23 @@ def post(self, request, *args, **kwargs):
             status = 400
 
         return Response(resp, status=status)
+
+
+class PersonnelCodeView(APIView):
+
+    """ PersonnelCodeView Interaction
+        GET - returns a list of identities with personnel codes
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def get(self, request, *args, **kwargs):
+        status = 200
+        resp = {"results": []}
+
+        identities = utils.search_identities(
+            {"details__has_key": "personnel_code"})
+
+        for identity in identities:
+            resp["results"].append(identity['details'].get('personnel_code'))
```

Comment:
I think that we should maybe put these codes in a set rather, to avoid duplicates.